### ISSUE TITLE
fix(node colours): Revisit the colour theming for nodes

### ISSUE THREAD

**Issue by simonwinter** [stencila/stencila#2126]
### Summary

Currently we're relying on very generic tailwind colours to set our node background colours etc.

```ts
const nodeColours = (name: string) => ({
  colour: colours[name][100],
  borderColour: colours[name][200],
})
```

In general, the design requires more variation than this. We need to review the available design tokens.

**Examples:**

_Note: I think there may be some colour mistakes here - black for block heading, incorrect purple on icon when not needed._

![CleanShot 2024-03-14 at 12 05 07](https://github.com/stencila/stencila/assets/1104537/5cb1ec90-1923-4324-a961-26a5765500eb)

![CleanShot 2024-03-14 at 12 05 32](https://github.com/stencila/stencila/assets/1104537/bfe23e2e-69a1-452d-8ade-5a6af2ac599c)

![CleanShot 2024-03-14 at 12 05 58](https://github.com/stencila/stencila/assets/1104537/347dbc2e-3341-4ef7-8d67-06b0cec31545)

**Comment by nokome** [comment]
Yes, agreed. Someone needs to go through and finish populating the `nodeTypeUIMap` with specific `colour` and `borderColour` hex (?) colours. The `nodeColours` func was just an expedient gap filler until the final colors were chosen in the Figma designs. 

Having said that, I'm not sure we have final colour choices and there are some inconsistencies in the latest designs (e.g. `CodeChunk` is a green, whereas its inline equivalent, a `CodeExpression` is a blue). So as part of this lets review all the colors used.
